modify rule S3616: add exception for `switch(true)` (#1070)
SonarSource/rspec: rules/S3616/javascript/rule.adoc

[source,javascript]
----

> switch (a) {

Author: :)

}
----

== Exceptions

> `Switch(true)` is a used construction when matching against expressions instead of values. Expressions in the `case` clauses will be evaluated before matching, meaning any expression that evaluates to `true` will be matched

Reviewer: ```suggestion
The `switch (true)` pattern is a common programming practice to match against expressions rather than literals. Expressions in the `case` clauses act as guards, i.e., they must evaluate to `true` if the execution is to continue in the branch.
```

[source,javascript]
----
function weekStatus (value) {

>   let day;

Reviewer: I would rename `value` to `day` and `day` to `status`.

  switch (true) {
    case (value === 'MON' || value === 'TUE' || value === 'WED' || value === 'THU' || value === 'FRI'):

>       day = "Week";

Reviewer: ```suggestion
      day = "Weekday";
```

>       status = "Weekday";

Reviewer: Just to be consistent, let's use single quote everywhere.

      break;
    case (day === 'SAT' || day === 'SUN'):

>       status = "Weekend";

Reviewer: ```suggestion
      status = 'Weekend';
```

      break;
  }
  return day;

> }

Reviewer: Nice example!

Reviewer: Nice copy-paste then :)
